mgmt/src/models/external/overlay/vpcpeering.rs

```diff
@@ -125,4 +257,489 @@ impl VpcPeeringTable {
             .values()
             .filter(move |p| p.left.name == vpc || p.right.name == vpc)
     }
+
+    fn vpcs(&self) -> HashSet<&String> {
```

Comment:
I believe this adds unnecessary code and steps.
As part of the overlay validation (https://github.com/githedgehog/dataplane/blob/pr/qmonnet/nat-validation/mgmt/src/models/external/overlay/mod.rs#L52) we collect all of the peerings where a VPC participates and add them to the VPC itself (see https://github.com/githedgehog/dataplane/blob/f42cd90d2f47e464b45a96c71ddecf4df5bb246e/mgmt/src/models/external/overlay/vpc.rs#L80). Can't you just validate the peerings once collected in the VPC to avoid this ?

Reply:
I can, but this means that we need to pass the `vpc_table` from the outer `Overlay` to the `validate()` function for the `peering_table()`. I wasn't sure that was the best approach, that's why I had recomputed the set separately.